mllib/src/main/scala/org/apache/spark/mllib/clustering/BisectingKMeansModel.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.clustering
+
+import breeze.linalg.{Vector => BV, norm => breezeNorm}
+
+import org.apache.spark.Logging
+import org.apache.spark.annotation.Since
+import org.apache.spark.api.java.JavaRDD
+import org.apache.spark.mllib.linalg.Vector
+import org.apache.spark.rdd.RDD
+
+/**
+ * This class is used for the model of the bisecting kmeans
+ *
+ * @param node a cluster as a tree node
+ */
+@Since("1.6.0")
+class BisectingKMeansModel @Since("1.6.0") (
+    @Since("1.6.0") val node: BisectingClusterNode
+  ) extends Serializable with Logging {
+
+  @Since("1.6.0")
+  def getClusters: Array[BisectingClusterNode] = this.node.getLeavesNodes
+
+  @Since("1.6.0")
+  def getCenters: Array[Vector] = this.getClusters.map(_.center)
+
+  /**
+   * Predicts the closest cluster by one point
+   */
+  @Since("1.6.0")
+  def predict(vector: Vector): Int = {
```

Comment:
* We need to do this from root to leaf so we don't need to compare the point with all leaf clusters.